### ISSUE TITLE
invocation of user-function at top level appears to halt rendering

### ISSUE THREAD

**Issue by NeilMacMullen** [scriban/scriban#317]
The output of 
```
{{func self;ret $0;end}} 


{{func viaFunction(p) 
[1,2,3] | array.each @p
end}} 


{{func inFunction 
[1,2,3] | array.each @self
end}} 

via function:  {{viaFunction @self}} EOL
in function:   {{inFunction}} EOL
at top: {{[1,2] | array.each @self}} EOL

```

is 

```
via function:  [1, 2, 3] EOL
in function:   [1, 2, 3] EOL
at top: [1, 2]
```
Note missing "EOL" at the end.

**Comment by NeilMacMullen** [comment]
No errors are thrown during parsing or rendering - the output just stops.   It looks like a rendering rather than parsing issue since inserting garbage code after the final line still results in a parser error.

**Comment by xoofx** [comment]
Yeah, I think I know where the issue is, let me try to push a fix...

**Comment by xoofx** [comment]
It should be fixed by commit cceabef5cd2e9073d28f0583bd03aa6bed6d402f
